lib/new_relic/harvest/collector/protocol.ex

```diff
@@ -92,14 +92,10 @@ defmodule NewRelic.Harvest.Collector.Protocol do
   end
 
   defp parse_http_response({:ok, %{status_code: 200, body: body}}, _params) do
-    case Jason.decode(body) do
-      {:ok, response} ->
-        {:ok, response}
-
-      {:error, jason_exception} ->
-        NewRelic.log(:error, "Bad collector JSON: #{Exception.message(jason_exception)}")
-        {:error, :bad_collector_response}
-    end
+    {:ok, NewRelic.JSON.decode!(body)}
+  rescue e ->
+    NewRelic.log(:error, "Bad collector JSON: #{Exception.format_banner(:error, e)}")
+    {:error, :bad_collector_response}
```

Comment:
Same here, use `decode/1`...

Reply:
The reason I switched here was that the error tuple that is returned from the native version was less descriptive, like `{:invalid_sequence, 60}` or something like that, and the prior handling in the `NewRelic.log` errored.

To keep the same logging, this is the way to go. Not sure what is preferred here.

Reply:
```suggestion
    {:ok, NewRelic.JSON.decode!(body)}
  rescue error ->
    NewRelic.log(:error, "Bad collector JSON: #{Exception.message(error)}")
    {:error, :bad_collector_response}
```